src/app/util/ember-compatibility-functions.cpp

```diff
@@ -94,5 +97,128 @@ bool ServerClusterCommandExists(chip::ClusterId aClusterId, chip::CommandId aCom
     return emberAfContainsServer(aEndPointId, aClusterId);
 }
 
+bool ServerClusterAttributeExists(chip::EndpointId aEndPointId, chip::ClusterId aClusterId, chip::AttributeId aAttributeId)
+{
+    // TODO: Currently, we are using cluster catalog from the ember library, this should be modified or replaced after several
+    // updates to Commands.
+    return emberAfReadServerAttribute(aEndPointId, aClusterId, aAttributeId, nullptr, 0) == EMBER_ZCL_STATUS_SUCCESS;
+}
+
+CHIP_ERROR ReadSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVWriter & aWriter)
+{
+    ChipLogDetail(DataManagement,
+                  "Received Cluster Command: Cluster=%" PRIx16 " NodeId=0x" ChipLogFormatX64 " Endpoint=%" PRIx8 " FieldId=%" PRIx8
+                  " ListIndex=%" PRIx8,
+                  aClusterInfo.mClusterId, ChipLogValueX64(aClusterInfo.mNodeId), aClusterInfo.mEndpointId, aClusterInfo.mFieldId,
+                  aClusterInfo.mListIndex);
+    uint8_t data[256];
```

Comment:
Added a TODO for this.

Choose 256 since it can hold values as large as OCTET_STRING, which should meet our current need, and can hold some small lists.

Reply:
Sadly, I have converted ember to use `uint16_t` from `uint8_t` for attributes size because that was needed by the operation credential cluster IIRC correctly. So I don't think it will meet our needs.

Reply:
Now using a bufferwith size EMBER_AF_MAXIMUM_SEND_PAYLOAD_LENGTH, which is also the buffer size used in ember library.

Reply:
I thought ember was using `ATTRIBUTE_LARGEST` which is a generated value ? https://github.com/project-chip/connectedhomeip/blob/30daae2b38d8c07c5797f073942113b774f8f299/src/app/util/attribute-table.cpp#L259